pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -385,6 +397,16 @@ func (host *BareMetalHost) ClearError() (dirty bool) {
 	return dirty
 }
 
+// RecordBoot records the current time as the LastBooted time.
+func (host *BareMetalHost) RecordBoot() {
+	now := metav1.Now()
+	host.Status.LastBooted = &now
+	if !host.Status.PendingRebootSince.IsZero() &&
```

Comment:
If PendingRebootSince is nil what will this do? Does IsZero() handle that?

Reply:
Yes, it's handled: https://github.com/metal3-io/baremetal-operator/blob/master/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/time.go#L67